common/persistence/data_interfaces.go

```diff
@@ -1204,6 +1205,52 @@ type (
 		SaveClusterMetadata(ctx context.Context, request *SaveClusterMetadataRequest) (bool, error)
 		DeleteClusterMetadata(ctx context.Context, request *DeleteClusterMetadataRequest) error
 	}
+
+	// HistoryTaskQueueManager is responsible for managing a queue of internal history tasks. It is currently unused,
+	// but we plan on using this to implement a DLQ for history tasks. This is called a history task queue manager, but
+	// the actual history task queues are not managed by this object. Instead, this object is responsible for managing
+	// a generic queue of history tasks (which is what the history task DLQ will be).
+	HistoryTaskQueueManager struct {
```

Comment:
hmm i guess we probably still need an interface for this. We will need to create metrics client, retryable client, rate limited client, fault injection wrapper for it. 
and do you plan to have a separate PR to adding those impls and wire up with the persistence factory?

Reply:
Added a TODO